plugins/cpp/parser/src/clangastvisitor.h

```diff
@@ -1278,10 +1281,95 @@ class ClangASTVisitor : public clang::RecursiveASTVisitor<ClangASTVisitor>
     return getVisibility(decl_->getAccess());
   }
 
-  bool isWritten(const clang::VarDecl* vd_) const
+  bool isWritable(const clang::QualType& param) const
   {
-    // TODO
-    return true;
+    const clang::Type* typePtr = param.getTypePtr();
+
+    return
+      ((typePtr->isReferenceType() || typePtr->isPointerType()) &&
+       !typePtr->getPointeeType().isConstQualified()) ||
+      typePtr->isRValueReferenceType();
+  }
+
+  bool isWritten(const clang::Expr* expr_) const
+  {
+    while (expr_)
+    {
+      clang::ASTContext::DynTypedNodeList parents
+        = _astContext.getParents(*expr_);
+
+      const clang::ast_type_traits::DynTypedNode& parent = parents[0];
```

Comment:
Is it always true, that `parents` variable always non-empty? `DynTypedNodeList` has an `empty()` function.